src/main/java/nextstep/courses/domain/session/FreeSession.java

```diff
@@ -1,22 +1,50 @@
 package nextstep.courses.domain.session;
 
 import nextstep.courses.domain.session.image.CoverImage;
+import nextstep.payments.domain.Payment;
 import nextstep.users.domain.NsUser;
-import nextstep.users.domain.Users;
 
 public class FreeSession extends Session {
+    public static final Long FREE_FEE = 0L;
 
     public FreeSession(Long sessionId, SessionPeriod sessionPeriod,
-                       CoverImage coverImage, SessionStatusEnum sessionStatus, Users users) {
-        super(sessionId, sessionPeriod, coverImage, sessionStatus, users);
+                       CoverImage coverImage, SessionStatusEnum sessionStatus) {
+        super(sessionId, sessionPeriod, coverImage, sessionStatus);
+        this.fee = FREE_FEE;
     }
 
     @Override
-    public void enrollStudent(NsUser user) {
+    public void enrollStudent(NsUser user, Payment payment) {
+        assertAllConditions(user, payment);
+        registerSession(user, payment);
+    }
+
+    private void assertAllConditions(NsUser user, Payment payment) {
         if (!this.isSessionOpened()) {
-            throw new IllegalArgumentException(SESSION_IS_NOT_OPENED);
+            throw new IllegalArgumentException(SESSION_NOT_OPENED);
+        }
+
+        if (isPaymentAmountNotMatching(payment, fee)) {
+            throw new IllegalArgumentException(PAYMENT_NOT_MATCHING);
+        }
+
+        if (hasAlreadyEnrolled(user, this)) {
+            throw new IllegalArgumentException(ENROLLMENT_ALREADY_DONE);
         }
+    }
+
+    private boolean isPaymentAmountNotMatching(Payment payment, Long fee) {
+        return !payment.isSamePriceWith(fee);
+    }
 
+    private boolean hasAlreadyEnrolled(NsUser nsUser, Session session) {
+        return nsUser.hasEnrolledSession(session);
+    }
```

Comment:
메서드 내부에서 this로 바로 전달해도 될 것 같습니다.
```suggestion
    private boolean hasAlreadyEnrolled(NsUser nsUser) {
        return nsUser.hasEnrolledSession(this);
    }
```